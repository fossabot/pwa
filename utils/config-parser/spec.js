const JSONConfig = require('./index');

describe('Utils', () => {
  describe('config-parser', () => {
    it('should replace a value within some object property inside object', () => {
      const config = {
        global: {
          foo: 'bar',
          baz: 'faz',
        },
        someOther: [
          {
            test: '1234',
            test2: '$.global.foo',
          },
        ],
      };
      const expected = {
        global: {
          foo: 'bar',
          baz: 'faz',
        },
        someOther: [
          {
            test: '1234',
            test2: 'bar',
          },
        ],
      };

      const result = new JSONConfig(config).parse();
      expect(result).toMatchObject(expected);
      expect(result).toMatchSnapshot();
    });

    it('should replace a value within some array value', () => {
      const config = {
        global: {
          foo: 'bar',
          baz: 'faz',
        },
        content: [
          '$.global.foo',
          '$.global.baz',
        ],
      };
      const expected = {
        global: {
          foo: 'bar',
          baz: 'faz',
        },
        content: [
          'bar',
          'faz',
        ],
      };

      const result = new JSONConfig(config).parse();
      expect(result).toMatchObject(expected);
      expect(result).toMatchSnapshot();
    });

    it('should not change numbers', () => {
      const config = {
        someNumber: 4,
        someArray: [
          5,
        ],
      };

      const expected = {
        someNumber: 4,
        someArray: [
          5,
        ],
      };

      const result = new JSONConfig(config).parse();
      expect(result).toMatchObject(expected);
      expect(result).toMatchSnapshot();
    });

    it('should not change arrays', () => {
      const config = {
        someNumber: 4,
        someArray: [
          [
            6,
            '$.someNumber',
          ],
        ],
      };

      const expected = {
        someNumber: 4,
        someArray: [
          [
            6,
            4,
          ],
        ],
      };

      const result = new JSONConfig(config).parse();
      expect(result).toMatchObject(expected);
      expect(result).toMatchSnapshot();
    });

    it('should return empty object if no config is set', () => {
      const result = new JSONConfig().parse();
      expect(result).toMatchObject({});
      expect(result).toMatchSnapshot();
    });
  });
});
