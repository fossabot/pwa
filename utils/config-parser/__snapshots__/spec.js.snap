// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils ConfigParser should not change arrays 1`] = `
Object {
  "someArray": Array [
    Array [
      6,
      4,
    ],
  ],
  "someNumber": 4,
}
`;

exports[`Utils ConfigParser should not change numbers 1`] = `
Object {
  "someArray": Array [
    5,
  ],
  "someNumber": 4,
}
`;

exports[`Utils ConfigParser should replace a value within some array value 1`] = `
Object {
  "content": Array [
    "bar",
    "faz",
  ],
  "global": Object {
    "baz": "faz",
    "foo": "bar",
  },
}
`;

exports[`Utils ConfigParser should replace a value within some object property inside object 1`] = `
Object {
  "global": Object {
    "baz": "faz",
    "foo": "bar",
  },
  "someOther": Array [
    Object {
      "test": "1234",
      "test2": "bar",
    },
  ],
}
`;

exports[`Utils ConfigParser should return empty object if no config is set 1`] = `Object {}`;
